pkg/reloader/reloader.go

```diff
@@ -203,6 +201,33 @@ func (r *Reloader) apply(ctx context.Context) error {
 	return nil
 }
 
+// write output file, extracting if necessary
+func (r *Reloader) writeOutputFile(b []byte) error {
```

Comment:
This method name hides few things, maybe worth to unpack this in `apply` method itself? But not a blocker.